pkg/rpc/context_test.go

```diff
@@ -1272,10 +1274,16 @@ func TestHeartbeatHealthTransport(t *testing.T) {
 		return nil
 	})
 
+	// TODO(baptist): Better understand when this happens. It appears we can get
+	// spurious connections to other tests on a stress run. This has been
+	// happening for a while, but only comes out rarely when this package is
+	// stressed. This test is very aggressive since it is calling GRPCDialNode in
+	// a busy loop for 50ms.
```

Comment:
I think this is a known problem. Test servers bind to port 0 to let the OS pick a random unused port, but as soon as we close it then a different test can be randomly assigned that very same port number, so if we later try to connect to that port we get unexpected results. This typically causes problems when we attempt to restart a stopped server reusing the same port number it was previously assigned, but someone else bound to it in the meanwhile.